searx/engines/tavily.py

```diff
@@ -0,0 +1,246 @@
+supported."""
+
+days: int = 3
+"""The number of days back from the current date to include in the search results.
+This specifies the time frame of data to be retrieved. Please note that this
+feature is only available when using the ``news`` search topic. Default is 3."""
+
+max_results: int = 5
+"""The maximum number of search results to return. Default is 5."""
+
+include_answer: bool = True
+"""Include a short answer to the original query, generated by an LLM based on Tavily's
+search results."""
+
+include_images: bool = False
+"""Include a list of query-related images in the response. Creates an infobox
+with the first image (as far as there are any images in the response) and the answer,
+if ``include_answer`` is also enabled.
+"""
+
+include_image_descriptions: bool = False
+"""When ``include_images`` is set to True, this option adds descriptive text for
+each image."""
+
+include_domains: list[str] = []
+"""A list of domains to specifically include in the search results. Default
+is ``[]``, which includes all domains."""
+
+exclude_domains: list[str] = []
+"""A list of domains to specifically exclude from the search results. Default
+is ``[]``, which doesn't exclude any domains.
+"""
+
+
+def request(query, params):
+
+    data = {
+        "query": query,
+        "api_key": api_key,
+        "search_depth": search_depth,
+        "topic": topic,
+        "time_range": params["time_range"],
+        "max_results": max_results,
+        "include_images": include_images,
+        "include_domains": include_domains,
+        "exclude_domains": exclude_domains,
+    }
+
+    if include_images:
+        data["include_image_descriptions"] = include_image_descriptions
+
+    if topic == "general":
+        data["include_answer"] = include_answer
+
+    elif topic == "news":
+        data["days"] = days
+
+    params["url"] = search_url
+    params["method"] = "POST"
+    params["headers"]["Content-type"] = "application/json"
+    params["data"] = dumps(data)
+
+    return params
+
+
+def response(resp):
+    results = []
+    data = resp.json()
+
+    for result in data.get("results", []):
+        results.append(
+            {
+                "title": result['title'],
+                "url": result["url"],
+                "content": f"[{gettext('ai')}] {result['content']}",
+                "publishedDate": _parse_date(result.get("published_date")),
+            }
+        )
+
+    img_list = data.get("images")
+    if img_list:
+        result = {
+            "infobox": f"Tavily [{gettext('ai')}]",
+            "img_src": img_list[0],
+        }
+
+        content = data.get("answer")
+        if isinstance(img_list[0], dict):
+            result["img_src"] = img_list[0]["url"]
+            img_caption = f"<i>{gettext('Image caption')}</i>: {img_list[0]['description']}"
+            if not content:
+                result["content"] = img_caption
+            else:
+                result["content"] = f"{content}<br/>{img_caption}"
```

Comment:
Do not use HTML markups in the results .. "separation of content structure and layout" 

- "structure" is given by the results their [types (I'm working on)](https://github.com/searxng/searxng/pull/4183) and their order.

- "layout" is done by the HTML templates & CSS

.. or in more practical terms: remember, we have also a JSON output. 

But for now leave it as is .. I will come back and review this PR next week / BTW thanks for your additions and corrections :+1: